### PR TITLE
Better click-dragging support for food + civilian cyborg tongs tool

### PR DESCRIPTION
**Pull request by codeJester27**
[feature][wiki]
## About the PR 
1. Click-dragging a food item onto another food item now uses the click-dragged food item on the other food item. So now, for example, dragging sugar onto dough makes sweet dough.
2. Civilian cyborgs now get a "tongs" tool that can be used on a food item to do the equivalent of using that food item in-hand. So now, for example, using tongs on a dough strip will twist the dough strip into a dough circle.
3. The "tongs" tool can also get food items out of food boxes, like egg cartons, or storage items, like boxes.
image

## Why's this needed? 
Lets civilian cyborgs make more dishes.

## Changelog

```changelog
(u)CodeJester
(+)Cyborgs can now click-drag food items onto each other to use them on each other. Try dragging sugar on dough to make sweet dough!
(+)Civilian cyborgs have a new "tongs" tool that can use a food item as if it were used in-hand. Try using tongs on a dough strip to make a dough circle!
(+)The "tongs" tool can also get food items out of containers, like egg cartons or boxes.
```

### CODE REVIEW

**Comment by mordent-goonstation** [comment]
Decent QoL change in terms of what it allows, but I think the implementation is a step in the wrong direction for specifically the cases it's intending to help.

I would prefer adding tools to civilian modules to cover the missing functionality.

**Comment by codeJester27** [comment]
Every kitchen appliance, to my knowledge, has a mousedrop proc (that usually calls `Attackby`) specifically to allow cyborgs to use items on them. Therefore this PR makes food items have the same consistent behavior, which keeps the click-drag style of being a cyborg.

As for the "drag food item on yourself to `AttackSelf`" behavior, I can't think of a tool that would fulfill every purpose like twisting a dough strip into a circle, mold sticky rice into rice balls, crunch a tortilla shell into tortilla chips, unwrap candy, unpeel bananas, etc..

**Comment by TDHooligan** [comment]
i'm a big fan of the idea, for both humans and cyborgs - for roleplay I can see a lot more stuff being done on tables with this PR, making chef a bit more showy overall.

**Comment by mordent-goonstation** [comment]
> Every kitchen appliance, to my knowledge, has a mousedrop proc (that usually calls `Attackby`) specifically to allow cyborgs to use items on them. Therefore this PR makes food items have the same consistent behavior, which keeps the click-drag style of being a cyborg.

Yep, I get the intent. Of the two parts, this one I think makes the most "sense".

> As for the "drag food item on yourself to `AttackSelf`" behavior, I can't think of a tool that would fulfill every purpose like twisting a dough strip into a circle, mold sticky rice into rice balls, crunch a tortilla shell into tortilla chips, unwrap candy, unpeel bananas, etc..

Without intending to be snarky, you PRed a magtractor thing that would achieve this a day before this PR that would have achieved the same goal; the "realism" aspect isn't *super* important.

A key direction I would like to steer towards is that non-civilian cyborgs should not be able to operate a kitchen basically as well as civilian ones. (I know non-civilian ones lack a couple of the tools, like the knife and the rolling pin, so can't do *everything* right now, but they're so close in functionality).

I think this PR heads away from that goal by making the kitchen *too* usable for non-civilian cyborgs.

**Comment by mordent-goonstation** [comment]
Much happier with the feature, will need someone who remembers how DM works to chime in on implementation so not explicitly approving.

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 1]
```diff
@@ -150,6 +150,12 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 				return
 
 			src.Eat(user,user)
+		else if (istype(W, /obj/item/tongs))
+			if (istype(src.loc, /obj/item/storage))
+				boutput(user, "You take [src] out of [src.loc].")
+				user.put_in_hand_or_drop(src)
+			else
+				src.AttackSelf(user)
```
Comment (frawhst) [inline code/modules/chemistry/tools/food_and_drink.dm:158]: should move the relevant code in `attack_self` to some other proc and then call that proc in `attack_self` and here; current implementation is likely fine but might cause weird issues down the line when something expects us to be using an item on itself in the usual way and we... aren't

note: a lot of existing code follows this pattern (mousedrop calling attackby, etc) which isn't ideal!

File: code/modules/chemistry/tools/food_and_drink.dm [hunk 2]
```diff
@@ -311,6 +317,11 @@ ABSTRACT_TYPE(/obj/item/reagent_containers/food/snacks)
 	afterattack(obj/target, mob/user, flag)
 		return
 
+	MouseDrop_T(obj/item/reagent_containers/food/snacks/O, mob/living/user)
+		if (istype(O) && istype(user) && in_interact_range(O, user) && in_interact_range(src, user))
+			return src.Attackby(O, user)
```
Comment (frawhst) [inline code/modules/chemistry/tools/food_and_drink.dm:322]: same as above but for `attackby`

**Comment by frawhst** [comment]
Oh wait, I didn’t read the above conversation closely enough. This is fine considering how the rest of kitchen code is set up.

**Review by frawhst — APPROVED** [review]

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**